src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 10;
+    private static final double INCREASE_ARRAY = 1.5;
+    private Object[] arraylist;
+    private int size;
+
+    public ArrayList() {
+        arraylist = new Object[ARRAY_SIZE];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == arraylist.length) {
+            grow();
+        }
+        arraylist[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAdd(index);
+        if (size == arraylist.length) {
+            grow();
+        }
+        System.arraycopy(arraylist, index, arraylist, index + 1, size - index);
+        arraylist[index] = value;
+        size++;
+    }
 
+    private void grow() {
+        Object[] newArray = new Object[(int) (arraylist.length * INCREASE_ARRAY)];
+        System.arraycopy(arraylist,0,newArray,0,size);
```

Comment:
please add whitespaces and don't resolve the comments if they are not fixed
you can ask questions if something is not clear

Reply:
ok, sorry, i just tried make correction as soon as possible, that's why i pass this System.arraycopy